src/nat/eval/runners/red_teaming_runner/runner.py

```
@@ -0,0 +1,866 @@
+        Returns:
+            A list of flat dictionaries, one per condition evaluation.
+        """
+        flat_results = []
+        evaluator_results = self._find_red_teaming_evaluator_results(results)
+
+        for scenario_id, result in evaluator_results.items():
+            for eval_output_item in result.eval_output_items:
+                item_id = eval_output_item.id
+                if not isinstance(eval_output_item, RedTeamingEvalOutputItem):
+                    raise ValueError("Expected RedTeamingEvalOutputItem, as an output to the red teaming evaluator,"
+                                     f"got {type(eval_output_item)}")
```

Ruff (0.14.8)

Prefer `TypeError` exception for invalid type

(TRY004)